### PR TITLE
Do the maven active profile substitution in bootstrap.yml for microservices

### LINKED ISSUE #4132: no cloud config in boostrap phase
I was about building our first 3.7 service at office and lost a couple of time by looking, why several settings not work....

if you start a JHipster application from war directly, or using docker....at the very first moment no bootstrap configuration is loaded. So setting SPRING_CLOUD_CONFIG_URI doesn't override anything, and finally a lot of things fail. In my case I don't get any connection to SQL db, because the config never arrives the app.

to reproduce that issue, just start a microservice using `java -jar` or docker, and watch the first line commiting a successful handshake with config server (like `s.n.www.protocol.http.HttpURLConnection  : sun.net.www.MessageHeader@7d3e86556 pairs: {GET /config/UAA/dev/master HTTP/1.1: null}`) is missing...

I guess this may have something to do with work on DefaultProfileUtil, which recently was done...

**edit:** copying bootstrap-prod.yml to bootstrap.yml is a dirty fix to that issue

### PR DESCRIPTION
**Pull request by PierreBesson**
fix #4132

### CODE REVIEW

**Comment by deepu105** [comment]
can you check if https://github.com/jhipster/generator-jhipster/pull/4185 has any impact here

File: generators/server/templates/src/main/resources/config/_bootstrap.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+spring:
```
Comment (deepu105) [inline generators/server/templates/src/main/resources/config/_bootstrap.yml]: you need to add this to the `profile_dev|prod.gradle` file in `processResources` section else value will not be replaced. not sure if anything is required for maven

**Comment by jdubois** [comment]
Shouldn't we just merge #4185 instead?

**Comment by deepu105** [comment]
@jdubois

**Comment by deepu105** [comment]
@jdubois not sure if #4185 alone will help here. @PierreBesson can you check if this is still an issue with #4185 as well

**Comment by PierreBesson** [comment]
This is to fix #4132 which is a different bug than what Deepu is talking about. Waiting for @xetys confirmation that it's also working for him.
@deepu, I will do the modification you requested tomorrow.

**Comment by xetys** [comment]
I am sorry, but testing with gradle led to the issue being still present to me. I tried both, merging this PR to tag v3.7.1 and to the current master, in both regeneration I get a

```
2016-09-22 13:45:28.166  WARN 9873 --- [  restartedMain] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/bar/default": Verbindungsaufbau abgelehnt; nested exception is java.net.ConnectException
```
where localhost:8888 is the spring default value, because it actually *is* overwritten in the profile specific bootstrap files. After changing the "#spring.profiles.active#" to "dev", it started to connect properly for gradle, bus was impassible to all kinds of `-Dspring.profiles.active´ or `--spring.profiles.active`

**Comment by deepu105** [comment]
Are you running it from the IDE?

Thanks & regards,
Deepu

**Comment by xetys** [comment]
tried with IntelliJ and `./gradlew` from cmd, both the same

**Comment by PierreBesson** [comment]
I just what @deepu105 told me, @xetys I hope that it will fix your issue.

**Comment by ruddell** [comment]
@PierreBesson I tested the fix and microservices generated with master can now connect to the registry on startup 👍

**Comment by PierreBesson** [comment]
Thank you everyone for helping me test this.

**Comment by jdubois** [comment]
Does this mean we should have a new release?

**Comment by deepu105** [comment]
Yes we can have one

**Comment by jdubois** [comment]
OK I will do a release next week probably... I'm starting a new gig tomorrow, so I won't have time before.

**Pull request merged**